src/main/java/org/opensrp/web/rest/PlanResource.java

```diff
@@ -372,5 +401,30 @@ public boolean isReturnCount() {
 		}
 
 	}
+
+	private PlanSearchBean createPlanSearchBean(boolean isTemplateParam, Integer pageNumber, Integer pageSize, String orderByType,
+			String orderByFieldName, String planStatus, Map<String, String> useContextFilters) {
+		PlanSearchBean planSearchBean = new PlanSearchBean();
+		planSearchBean.setExperimental(isTemplateParam);
+		if(pageNumber != null) {
+			planSearchBean.setPageNumber(pageNumber);
+		}
+		if(pageSize != null) {
+			planSearchBean.setPageSize(pageSize);
+		}
+		if (orderByType != null) {
+			planSearchBean.setOrderByType(PlanSearchBean.OrderByType.valueOf(orderByType));
+		}
+		if (orderByFieldName != null) {
+			planSearchBean.setOrderByFieldName(PlanSearchBean.FieldName.valueOf(orderByFieldName));
+		}
+		if (planStatus != null) {
+			planSearchBean.setPlanStatus(PlanDefinition.PlanStatus.valueOf(planStatus));
+		}
+		if (useContextFilters != null) {
+			planSearchBean.setUseContexts(useContextFilters);
+		}
```

Comment:
are the if statements required here

Reply:
Setting fields in PlanSearchBean only when it's not null.

Reply:
Removed if statements to optimize complexity.

Reply:
Added if statements, where the **valueOf** method is being called to, get enum constant.